src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -120,17 +164,34 @@ export class ResultsViewPageStore {
 
     readonly samples = remoteData({
         await: () => [
-            this.sampleIds
+            this.studyToSampleIds
         ],
-        invoke: async () => fetchSamples(this.sampleIds, this.studyId)
+        invoke: () => {
+            let sampleIdentifiers:SampleIdentifier[] = [];
+            _.each(this.studyToSampleIds.result, (sampleIds:{[sampleId:string]:boolean}, studyId:string)=>{
+                sampleIdentifiers = sampleIdentifiers.concat(Object.keys(sampleIds).map(sampleId=>({sampleId, studyId})));
+            });
+            return client.fetchSamplesUsingPOST({
+                sampleIdentifiers,
+                projection: "DETAILED"
+            });
+        }
     }, []);
 
-    readonly samplesWithoutCancerTypeClinicalData = remoteData({
+    readonly samplesWithoutCancerTypeClinicalData = remoteData<Sample[]>({
         await: () => [
-            this.sampleIds,
+            this.samples,
             this.clinicalDataForSamples
         ],
-        invoke: async () => fetchSamplesWithoutCancerTypeClinicalData(this.sampleIds, this.studyId, this.clinicalDataForSamples)
+        invoke: () => {
+            const sampleHasData:{[sampleUid:string]:boolean} = {};
+            for (const data of this.clinicalDataForSamples.result) {
+                sampleHasData[toSampleUuid(data.clinicalAttribute.studyId, data.entityId)] = true;
+            }
+            return Promise.resolve(this.samples.result.filter(sample=>{
```

Comment:
why Promise.resolve?

Reply:
invoke has to return a promise. when its async thats done automatically, but I wanted to not use async here as a way of denoting that no async stuff is actually done